sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -813,9 +811,11 @@ private Block BuildSwitchExpression(SwitchExpressionSyntaxWrapper switchExpressi
                 var armBlock = BuildExpression(arm.Expression, CreateBlock(currentBlock));
 
                 currentArmBlock = BuildArmBranch(arm, armBlock, currentArmBlock);
+
+                currentArmBlock = BuildExpression(switchExpressionSyntax.GoverningExpression, currentArmBlock);
```

Comment:
In order to be consistent and to ease the exploded graph analysis I've added the switch governing expression on each arm.